test/http_memprof/http_memprof.go

```diff
@@ -0,0 +1,278 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// this code is a utility to monitor an already running mixer instance, and such
+// does not run with regular tests, so disable deadcode warning on it.
+//nolint:deadcode
+package main
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	pb "github.com/datacommonsorg/mixer/internal/proto"
+	"github.com/google/pprof/profile"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"time"
+)
+
+var (
+	grpcAddr          = flag.String("grpc_addr", "127.0.0.1:12345", "Address of grpc server.")
+	profAddr          = flag.String("prof_addr", "http://localhost:6060", "Address of HTTP profile server.")
+	outFolderParent   = flag.String("out_folder", "http_memprof_out", "Folder to store output of memory profiles over HTTP")
+	resultCsvFilename = "results.csv"
+)
+
+func readProfile(filename string) (*profile.Profile, error) {
+	reader, err := os.Open(filename)
+	if err != nil {
+		return nil, errors.New("file could not be opened: " + filename)
+	}
+
+	prof, err := profile.Parse(reader)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse proto file: %v", filename)
+	}
+	return prof, err
+}
+
+// GetTotalSpaceAllocFromProfile reads the pprof compatible report file at the
+// given path and returns the total allocated space it recorded. Assumes that
+// the memory profile sample_index for alloc_space is 1
+func GetTotalSpaceAllocFromProfile(filename string) int64 {
+	// Modified from https://github.com/google/pprof/blob/c488b8fa1db3fa467bf30beb5a1d6f4f10bb1b87/internal/report/report.go
+	// =======================
+	// computeTotal computes the sum of the absolute value of all sample values.
+	// If any samples have label indicating they belong to the diff base, then the
+	// total will only include samples with that label.
+	computeTotal := func(prof *profile.Profile, value func(v []int64) int64) int64 {
+		var total, diffTotal int64
+		for _, sample := range prof.Sample {
+			var v int64
+			v = value(sample.Value)
+			if v < 0 {
+				v = -v
+			}
+			total += v
+			if sample.DiffBaseSample() {
+				diffTotal += v
+			}
+		}
+		if diffTotal > 0 {
+			total = diffTotal
+		}
+		return total
+	}
+	// sample index of alloc_space in neap profiles from http/net/pprof
+	const allocSpaceSampleIndex = 1
+
+	prof, err := readProfile(filename)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
you can let the function return (int64, err) and just do "return 0, err" here.  In general, golang passes the err out and only let the outmost function to handle.